File: backend/bibliotheque/src/main/java/com/upb/bibliotheque/entity/Reservation.java
package com.upb.bibliotheque.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "reservations")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor
public class Reservation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long reservationId;

    @ManyToOne
    @JoinColumn(name = "etudiant_id", nullable = false)
    private Etudiant etudiant;

    @ManyToOne
    @JoinColumn(name = "livre_id", nullable = false)
    private Livre livre;

    private LocalDateTime dateReservation = LocalDateTime.now();

    @Enumerated(EnumType.STRING)
    private StatutReservation statut = StatutReservation.EN_ATTENTE;
    private Integer positionFileAttente;

    public enum StatutReservation { EN_ATTENTE, CONFIRMEE, ANNULEE, REFUSEE }
}

File: shared/models/reservation.dart
class Reservation {
  final int? reservationId;
  final String dateReservation;
  final String statut;
  final int positionFileAttente;
  final Map<String, dynamic>? livre;

  Reservation({
    this.reservationId,
    required this.dateReservation,
    required this.statut,
    required this.positionFileAttente,
    this.livre,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) => Reservation(
    reservationId: json['reservationId'],
    dateReservation: json['dateReservation'] ?? '',
    statut: json['statut'] ?? 'EN_ATTENTE',
    positionFileAttente: json['positionFileAttente'] ?? 0,
    livre: json['livre'],
  );

  String get titreLivre => livre?['titre'] ?? 'Livre inconnu';
}